File: src/CITL.WebApi/Responses/ApiResponseCode.cs
namespace CITL.WebApi.Responses;

/// <summary>
/// Machine-readable numeric codes for the frontend to branch on.
/// </summary>
public static class ApiResponseCode
{
    /// <summary>Operation completed successfully.</summary>
    public const int Success = 1;

    /// <summary>Operation completed with a warning.</summary>
    public const int Warning = 0;

    /// <summary>Operation failed.</summary>
    public const int Error = -1;
}

File: src/CITL.Web/src/Shared/Types/ApiResponseCode.ts
/**
 * Machine-readable numeric codes matching CITL.WebApi.Responses.ApiResponseCode.
 * 1 = success, 0 = warning, -1 = error.
 */
export const ApiResponseCode = {
    Success: 1,
    Warning: 0,
    Error: -1,
} as const;

export type ApiResponseCode = (typeof ApiResponseCode)[ keyof typeof ApiResponseCode ];
